### ISSUE TITLE
Index documents on fields that contain lists of strings

### ISSUE THREAD

**Issue by brunogirin** [rluiten/elm-text-search#7]
I have a use case where it would be very useful if the index was able to use fields that hold lists of strings in addition to single strings and apply the same weight to all strings in the list. The documents in this use case look like:

``` elm
{ cid : String
, name : String
, synonyms : List String
}
```

So I'd like to index both the name and the synonyms and apply the same weight to all synonyms.

From a `SimpleConfig` / `Config` API, I would see it as adding a new option, something like:

``` elm
list_fields : List (doc -> List String, Float)
```

Which would be used like this:

``` elm
ElmTextSearch.new
{ ref = .cid
, fields = [ ( .name, 5.0 ) ]
, list_fields = [ ( .synonyms, 1.0 ) ]
}
```

**Comment by rluiten** [comment]
Short term you could just join the synonym fields into one String with String.join.
So create a new Document Type suitable and transform each one to that then add to index.
Its an extra step, but you can discard the transformed document as soon as its been added to the index no need to keep it so the overhead is very likely not bad at all.

I'll look a bit into adding direct support in the next few weekends I hope, it might get a bit fiddly in the plumbing I think.

**Comment by rluiten** [comment]
I had a couple hours spare and implemented this.
It didn't get to complex just makes the SimpleConfig and Config one field larger.
Added a new Config field  `listFields` to configure document fields of type List String.
Note the name change from your example as most identifiers are camel case with leading lower case.
Its published in 3.0.0.

**Comment by brunogirin** [comment]
Awesome, thanks!
